src/Sylius/Component/User/Model/User.php

```diff
@@ -547,6 +547,24 @@ public function unserialize($serialized): void
         ] = $data;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isEqualTo(\Symfony\Component\Security\Core\User\UserInterface $user): bool
+    {
+        if (!$user instanceof UserInterface) {
+            return false;
+        }
+
+        return $this->getPassword() === $user->getPassword()
+            && $this->getUsername() === $user->getUsername()
+            && $this->getEmail() === $user->getEmail()
```

Comment:
User email is never set in `ShopUserExampleFactory` (used in e.g. `I am a logged in customer` step) and because of that scenarios fail. We should either don't check it here or modify factory to set email on newly created shop user. The problem is, in `ShopUser` entity setting user email is [in fact setting it's customer email](https://github.com/Sylius/Sylius/blob/26ee97a41fb2add65966f16775e9dfb0349ebf6b/src/Sylius/Component/Core/Model/ShopUser.php#L68) 😄